syntax = "proto3";

package proto;
option go_package = "bus-project/proto";

message CANBusMessage {
    bytes can_id = 1;
    int32 dlc = 2;
    bytes data = 3;
    int64 timestamp = 4;
}

message CANBusMessageList {
    repeated CANBusMessage data = 1;
}

import "google/protobuf/empty.proto";

service CANBusService {
    rpc Send(CANBusMessage) returns (google.protobuf.Empty) {}
}
